SparrowImplicitLib/std/union.spr

```diff
@@ -0,0 +1,604 @@
+        using Type12 = t12
+        using Type13 = t13
+        using Type14 = t14
+        using Type15 = t15
+
+    fun get1(this: @Union15) = reinterpretCast(@this.Type1, this.data)
+    fun get2(this: @Union15) = reinterpretCast(@this.Type2, this.data)
+    fun get3(this: @Union15) = reinterpretCast(@this.Type3, this.data)
+    fun get4(this: @Union15) = reinterpretCast(@this.Type4, this.data)
+    fun get5(this: @Union15) = reinterpretCast(@this.Type5, this.data)
+    fun get6(this: @Union15) = reinterpretCast(@this.Type6, this.data)
+    fun get7(this: @Union15) = reinterpretCast(@this.Type7, this.data)
+    fun get8(this: @Union15) = reinterpretCast(@this.Type8, this.data)
+    fun get9(this: @Union15) = reinterpretCast(@this.Type9, this.data)
+    fun get10(this: @Union15) = reinterpretCast(@this.Type10, this.data)
+    fun get11(this: @Union15) = reinterpretCast(@this.Type11, this.data)
+    fun get12(this: @Union15) = reinterpretCast(@this.Type12, this.data)
+    fun get13(this: @Union15) = reinterpretCast(@this.Type13, this.data)
+    fun get14(this: @Union15) = reinterpretCast(@this.Type14, this.data)
+    fun get15(this: @Union15) = reinterpretCast(@this.Type15, this.data)
+
+package _Impl16
+    datatype Union16(t1, t2, t3, t4, t5, t6, t7, t8, t9, t10, t11, t12, t13, t14, t15, t16: Type)
+        data: StaticArray(Char, \
+            _maxSize( \
+                sizeOf(t16), \
+                _maxSize( \
+                    sizeOf(t15), \
+                    _maxSize( \
+                        sizeOf(t14), \
+                        _maxSize( \
+                            sizeOf(t13), \
+                            _maxSize( \
+                                sizeOf(t12), \
+                                _maxSize( \
+                                    sizeOf(t11), \
+                                    _maxSize( \
+                                        sizeOf(t10), \
+                                        _maxSize( \
+                                            sizeOf(t9), \
+                                            _maxSize( \
+                                                sizeOf(t8), \
+                                                _maxSize( \
+                                                    sizeOf(t7), \
+                                                    _maxSize( \
+                                                        sizeOf(t6), \
+                                                        _maxSize( \
+                                                            sizeOf(t5), \
+                                                            _maxSize( \
+                                                                sizeOf(t4), \
+                                                                _maxSize( \
+                                                                    sizeOf(t3), \
+                                                                    _maxSize(sizeOf(t1),sizeOf(t2)))))))))))))))))
```

Comment:
this doesn't look right.
maybe a better looking formula is:
`sizeOf(t1) _maxSize sizeOf(t2) _maxSize sizeOf(t3) ... _maxSize sizeOf(t16)`